__tests__/src/rules/has-valid-accessibility-state-test.js

```diff
@@ -0,0 +1,73 @@
+/* eslint-env jest */
+/**
+ * @fileoverview Describes the current state of a component to the user of an assistive technology.
+ * @author JP Driver
+ */
+
+// -----------------------------------------------------------------------------
+// Requirements
+// -----------------------------------------------------------------------------
+
+import { RuleTester } from 'eslint';
+import parserOptionsMapper from '../../__util__/parserOptionsMapper';
+import rule from '../../../src/rules/has-valid-accessibility-state';
+
+// -----------------------------------------------------------------------------
+// Tests
+// -----------------------------------------------------------------------------
+
+const ruleTester = new RuleTester();
+
+const propMustBeAnObject = {
+  message: 'accessibilityState must be an object',
+  type: 'JSXAttribute'
+};
+
+const invalidObjectKey = key => ({
+  message: `accessibilityState object: "${key}" is not a valid key`,
+  type: 'JSXAttribute'
+});
+
+const valueMustBeBoolean = key => ({
+  message: `accessibilityState object: "${key}" value is not a boolean`,
+  type: 'JSXAttribute'
+});
+
+const checkedMustBeBooleanOrMixed = {
+  message: `accessibilityState object: "checked" value is not either a boolean or 'mixed'`,
+  type: 'JSXAttribute'
+};
+
+ruleTester.run('has-valid-accessibility-state', rule, {
+  valid: [
```

Comment:
AFAICT the object can have several of these properties. Perhaps we should have a couple examples of these in our valid code examples and one example in the docs for this rule...